sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Azure.Core.Experimental.Primitives
+{
+    /// <summary>
+    /// Abstraction for a payload of binary data.
+    /// </summary>
+    public class BinaryData
```

Comment:
Why a class and not a struct?

Reply:
Ah woops, should be a struct for now at least.